Microsoft.Azure.CosmosRepository.Samples/OptimisticConcurrencyControl/Program.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System.Net;
+using Bogus;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.CosmosRepository;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using OptimisticConcurrencyControl;
+
+
+IRepository<BankAccount> BuildRepository(bool optimizeBandwidth)
+{
+    ConfigurationBuilder configuration = new();
+
+    ServiceProvider provider = new ServiceCollection().AddCosmosRepository(options =>
+        {
+            options.CosmosConnectionString = Environment.GetEnvironmentVariable("CosmosConnectionString");
+            options.DatabaseId = "optimistic-concurrency-control";
+            options.OptimizeBandwidth = optimizeBandwidth; // Must be false to receive the upto date etags back on update calls
+        })
+        .AddSingleton<IConfiguration>(configuration.Build())
+        .BuildServiceProvider();
+
+    IRepository<BankAccount> repository = provider.GetRequiredService<IRepository<BankAccount>>();
+
+    return repository;
+}
+
+Console.WriteLine("----------------------");
+await ConcurrencyWithOptimizeBandwidthOff();
+Console.WriteLine("----------------------");
+await ConcurrencyWithOptimizeBandwidthOn();
+Console.WriteLine("----------------------");
+await ConcurrencyWithOptimizeBandwidthOnTrapDemo();
+Console.WriteLine("----------------------");
+
+async Task ConcurrencyWithOptimizeBandwidthOff()
+{
+    Console.WriteLine($"Optimized bandwidth OFF.");
+    IRepository<BankAccount> repository = BuildRepository(false);
+
+    BankAccount currentBankAccount = await repository.CreateAsync(new BankAccount()
+    {
+        Name = "Current Account",
+        Balance = 500.0
+    });
+
+    Console.WriteLine($"Created a bank account: {currentBankAccount}.");
+
+    currentBankAccount.Withdraw(500);
+    await repository.UpdateAsync(currentBankAccount);
+    currentBankAccount.Deposit(500);
+
+    Console.WriteLine($"Simulating a withdrawal of 500 that happened from another service. Remote State - {await repository.GetAsync(currentBankAccount.Id)}. Local State - {currentBankAccount}");
+
+    Console.WriteLine("Attempting to withdraw 250 from the bank account.");
+    try
+    {
+        Console.WriteLine("Attempting to withdraw 250 from the bank account.");
+        await repository.UpdateAsync(currentBankAccount.Id,
+            builder => builder.Replace(account => account.Balance, currentBankAccount.Balance - 250), etag: currentBankAccount.Etag);
+    }
+    catch (CosmosException exception)
```

Comment:
```suggestion
    catch (CosmosException exception) when (excepton.StatusCode = StatusCode)
```

Then you can drop the nested `if`